internal/collections/named.go

```diff
@@ -116,10 +96,13 @@ func (c *NamedCollectionNames) FindString(key string) []types.MatchData {
 
 func (c *NamedCollectionNames) FindAll() []types.MatchData {
 	var res []types.MatchData
-	for _, k := range c.collection.names {
+	// Iterates over all the MatchData and adds the key element also to the Key field (The key value may be the value that is matched,
+	// but it is still also the key of the pair and it is needed to print the matched var name)
+	for _, k := range c.collection.Map.FindAll() {
```

Comment:
I think you should be able to iterate over `c.collection.Map.data` directly

Reply:
Done, thank you!

Reply:
Is it done actually? Maybe you forgot to push?

Reply:
I did it in [3a8ed2c](https://github.com/corazawaf/coraza/pull/618/commits/3a8ed2ca446aa990ac70be9d4640aac73a4394e8), but it was restricted to `FindAll()` function. Now I moved to `c.collection.Map.data` also in `String()` even if it adds a bit of logic to figure out when to print the comma